internal/command/postgres/restart.go

```diff
@@ -81,25 +64,50 @@ func runRestart(ctx context.Context) error {
 			return fmt.Errorf("can't fetch allocations: %w", err)
 		}
 		return nomadRestart(ctx, vms)
-
 	case "machines":
-		flapsClient, err := flaps.New(ctx, app)
-		if err != nil {
-			return fmt.Errorf("list of machines could not be retrieved: %w", err)
+		template := recipe.RecipeTemplate{
+			Name:         "Rolling restart",
+			RequireLease: true,
+			Operations: []recipe.Operation{
+				{
+					Name: "restart",
+					Type: recipe.OperationTypeMachine,
+					MachineCommand: recipe.MachineCommand{
+						Action: "restart",
+					},
+					HealthCheckSelector: recipe.HealthCheckSelector{
+						Name:  "role",
+						Value: "replica",
+					},
+				},
+				{
+					Name: "failover",
+					Type: recipe.OperationTypeHTTP,
+					HTTPCommand: recipe.HTTPCommand{
+						Method:   "GET",
+						Endpoint: "/commands/admin/failover/trigger",
+						Port:     5500,
+					},
+					HealthCheckSelector: recipe.HealthCheckSelector{
+						Name:  "role",
+						Value: "leader",
+					},
+				},
+				{
```

Comment:
This only works currently because we don't wait for health checks post-failover.  It's an interesting problem to address here.